s3torchconnector/src/s3torchconnector/_s3client/_s3client.py

```diff
@@ -32,11 +32,12 @@ def _identity(obj: Any) -> Any:
 
 
 class S3Client:
-    def __init__(self, region: str, endpoint: str = None):
+    def __init__(self, region: str, endpoint: str = None, user_agent: UserAgent = None):
         self._region = region
         self._endpoint = endpoint
         self._real_client = None
         self._client_pid = None
+        self._user_agent = user_agent or UserAgent()
```

Comment:
Are we expecting to mutate the user agent after it's been passed to an S3 client? If not this area can be simplified a lot